pkg/storegateway/bucket_test.go

```diff
@@ -1564,9 +1568,12 @@ func TestBucketStore_Series_OneBlock_InMemIndexCacheSegfault(t *testing.T) {
 		chunkPool:            chunkPool,
 	}
 
+	srv, err := newBucketStoreTestServer(store)
```

Comment:
Everywhere this is used it has these two post setup calls
```go
	require.NoError(t, err)
	t.Cleanup(srv.Close)
```

What do you think about adding `*testing.T` argument to `newBucketStoreTestServer` and at least moving the `require.NoError` there? Moving the cleanup is a bit more difficult since `querySeries` requires a func it can defer as opposed to doing it at the end of the test like the rest of the call sites.

Reply:
Makes sense. Done here: https://github.com/grafana/mimir/pull/4179/commits/c7082b24ec93515995517ad0c271eeb3ea04fb49